src/server.rs

```diff
@@ -365,13 +365,29 @@ impl Server {
         middleware_type: &MiddlewareType,
         route: &str,
         function: FunctionInfo,
+        http_method: Option<HttpMethod>,
     ) {
+
+        let mut endpoint = route.to_string();
+
+        if let Some(method) = http_method {
+
+            endpoint = method.to_string().to_owned();
+
+            if !route.starts_with('/') {
+                endpoint.push('/');
+            }
```

Comment:
shouldn't it be done at the fundamental level somewhere?

Reply:
this is just for the hashmap key. we are not exposing this.